idp/static/static.go

```diff
@@ -76,13 +81,28 @@ func NewIdentityProvider(p Params) idp.IdentityProvider {
 	if p.Description == "" {
 		p.Description = p.Name
 	}
-	return &identityProvider{params: p}
+	var matchEmailAddr *regexp.Regexp
+	if p.MatchEmailAddr != "" {
+		var err error
+		matchEmailAddr, err = regexp.Compile(p.MatchEmailAddr)
+		if err != nil {
+			// if the email address matcher doesn't compile log the error but
+			// carry on. A regular expression that doesn't compile also doesn't
+			// match anything.
+			logger.Errorf("cannot compile match-email-addr regular expression: %s", err)
```

Comment:
Should this constructor fail and return an error if the configuration is invalid?

Reply:
my thinking is in the comment, but I decided it wasn't worth failing for, instead I decided that nothing can match an invalid regular expression.